pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -217,6 +240,13 @@ func findCustomObjectByName(template api.CustomObject, objects []api.CustomObjec
 }
 
 func getEnvTemplate(cr *api.KieApp) (envTemplate api.EnvTemplate, err error) {
+
+	if !isImmutable(cr) && cr.Spec.Objects.Console == nil {
+		cr.Spec.Objects.Console = &api.ConsoleObject{
+			KieAppObject: api.KieAppObject{},
+		}
+	}
+
```

Comment:
I think this makes more sense in the SetDefaults() method and should be applied to the `specApply` "defaulting" object.

We assign defaults to `status.applied` now instead of directly to `spec`.

Reply:
I think I tired to add there but got some null pointer, will try again.

Reply:
should assign to specApply